openupgrade_scripts/scripts/calendar/15.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,58 @@
+from openupgradelib import openupgrade
+
+_model_renames = [
+    ("calendar.contacts", "calendar.filters"),
+]
+
+_table_renamed = [
+    ("calendar_contacts", "calendar_filters"),
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_models(env.cr, _model_renames)
+    openupgrade.rename_tables(env.cr, _table_renamed)
+    openupgrade.logged_query(
+        env.cr,
+        """
+            ALTER TABLE calendar_filters
+            ADD COLUMN IF NOT EXISTS partner_checked BOOLEAN;
+            UPDATE calendar_filters
+            SET partner_checked = TRUE;
+        """,
+    )
+
+    openupgrade.convert_field_to_html(
+        env.cr, "calendar_event", "description", "description"
+    )
+
+    openupgrade.rename_fields(
+        env,
+        [
+            ("calendar.recurrence", "calendar_recurrence", "fr", "fri"),
+            ("calendar.recurrence", "calendar_recurrence", "mo", "mon"),
+            ("calendar.recurrence", "calendar_recurrence", "sa", "sat"),
+            ("calendar.recurrence", "calendar_recurrence", "su", "sun"),
+            ("calendar.recurrence", "calendar_recurrence", "th", "thu"),
+            ("calendar.recurrence", "calendar_recurrence", "tu", "tue"),
+            ("calendar.recurrence", "calendar_recurrence", "we", "wed"),
+        ],
+    )
+
+    # change the value of 'weekday' in the calendar_recurrence table
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE calendar_recurrence
+            SET weekday = CASE WHEN weekday = 'FR' THEN 'FRI'
+                               WHEN weekday = 'MO' THEN 'MON'
+                               WHEN weekday = 'SA' THEN 'SAT'
+                               WHEN weekday = 'SU' THEN 'SUN'
+                               WHEN weekday = 'TH' THEN 'THU'
+                               WHEN weekday = 'TU' THEN 'TUE'
+                               WHEN weekday = 'WE' THEN 'WED'
+                               ELSE NULL
+                               END
```

Comment:
@hoangtiendung070797 code chắc cú thì phải như này, trường hợp có thêm tùy chọn khác thì cũng không lỗi
```suggestion
                               END
        WHERE weekday IN ('FR', 'MO', 'SA', 'SU', 'TH', 'TU', 'WE')
```

Reply:
@royleviindoo dạ vâng anh